### PR TITLE
Display total count instead of percent and order by count

### LINKED ISSUE #45: Results of a votation
- display the candidates in order from the ones who get the more votes to the less ones
- display the number of votes received (not the %). Ex : 44/234

Ex:
candidate 3  --------------------------------------------    120/234
candidate 1  ---------------------------------                   95/234
candidate 5  --------------------------                            78/234
candidate 2  ----------------                                        50/234
candidate 4  ---------                                                 35/234

### PR DESCRIPTION
**Pull request by PascalinDe**
Display election results as total count instead of percent and order by count. Adds tests for the results.

Closes #45

### CODE REVIEW

File: web/frontend/src/pages/form/GroupedResult.tsx [hunk 1]
```diff
@@ -124,9 +124,15 @@ const GroupedResult: FC<GroupedResultProps> = ({ rankResult, selectResult, textR
           const select = element as SelectQuestion;
 
           if (selectResult.has(id)) {
-            res = countSelectResult(selectResult.get(id)).resultsInPercent.map((percent, index) => {
-              return { Candidate: select.Choices[index], Percentage: `${percent}%` };
-            });
+            res = countSelectResult(selectResult.get(id))
+              .map(([, totalCount], index) => {
```
Comment (ineiti) [inline web/frontend/src/pages/form/GroupedResult.tsx:128]: The linter doesn't complain? I thought you'd have to write `.map(([_, totalCount]` here.

File: web/frontend/src/pages/form/GroupedResult.tsx [hunk 2]
```diff
@@ -124,9 +124,15 @@ const GroupedResult: FC<GroupedResultProps> = ({ rankResult, selectResult, textR
...
+                return {
+                  Candidate: select.Choices[index],
+                  TotalCount: totalCount,
+                  NumberOfBallots: selectResult.get(id).length,
```
Comment (ineiti) [inline web/frontend/src/pages/form/GroupedResult.tsx:132]: Question: what is `NumberOfBallots` here?
- the total ballots cast (including multi ballot cast)
- the total number of voters (with multi ballot cast removed)
- the total number of valid ballots (with multi ballot and invalid ballots removed)

I think both 2 and 3 are OK, but it should be mentioned somewhere on the page which it is.

File: web/frontend/src/pages/form/components/utils/countResult.ts [hunk 3]
```diff
@@ -42,31 +42,20 @@ const countRankResult = (rankResult: number[][], rank: RankQuestion) => {
...
-
-  const results = selectResult.reduce((a, b) => {
-    return a.map((value, index) => {
-      const current = value + b[index];
-
-      if (current >= max) {
-        max = current;
-      }
-      return current;
+  let results: [string, number][] = [];
```
Comment (ineiti) [inline web/frontend/src/pages/form/components/utils/countResult.ts:45]:
```suggestion
  const results: [string, number][] = [];
```

`push` and `pop` are allowed on `const`. But this makes sure that you don't re-assign it a different array.

File: web/frontend/src/pages/form/components/utils/countResult.ts [hunk 4]
```diff
@@ -42,31 +42,20 @@ const countRankResult = (rankResult: number[][], rank: RankQuestion) => {
...
+  selectResult
+    .reduce(
+      (tally, currBallot) => tally.map((currCount, index) => currCount + currBallot[index]),
+      new Array(selectResult[0].length).fill(0)
+    )
+    .forEach((totalCount) => {
+      results.push([
+        (Math.round((totalCount / selectResult.length) * 100 * 100) / 100).toFixed(2).toString(),
+        totalCount,
+      ]);
```
Comment (ineiti) [inline web/frontend/src/pages/form/components/utils/countResult.ts:56]: Or remove the `let result` altogether and do something like:

```suggestion
  return selectResult
    .reduce(
      (tally, currBallot) => tally.map((currCount, index) => currCount + currBallot[index]),
      new Array(selectResult[0].length).fill(0)
    )
    .map((totalCount) => 
      [
        (Math.round((totalCount / selectResult.length) * 100 * 100) / 100).toFixed(2).toString(),
        totalCount,
      ]);
```

File: web/frontend/tests/json/evoting/forms/canceled.json [hunk 5]
```diff
@@ -42,6 +42,43 @@
...
+          "riJFjw0q"
+        ],
+        "Subjects": [],
+        "Selects": [
+          {
+            "ID": "riJFjw0q",
+            "Title": {
+              "En": "CMYK",
+              "Fr": "CMJN",
+              "De": "CMYK"
```
Comment (ineiti) [inline web/frontend/tests/json/evoting/forms/canceled.json:63]: Joke: Wir haben keine Übersetzung dafür? Sowas... "CMGS"?

**Review by ineiti — APPROVED** [review]
Some comments, but nothing bad. As we're getting to the end of it, I'm also OK if you ignore my comments.

File: web/frontend/src/pages/form/GroupedResult.tsx (hunk 1 above)
Comment (PascalinDe) [reply web/frontend/src/pages/form/GroupedResult.tsx:128]: no, the linter is quite happy like that and actually made me change it to this :woman_shrugging:

File: web/frontend/src/pages/form/GroupedResult.tsx (hunk 2 above)
Comment (PascalinDe) [reply web/frontend/src/pages/form/GroupedResult.tsx:132]: it's the number of valid ballots for the 'select' type of form as send back by the server

File: web/frontend/tests/json/evoting/forms/canceled.json (hunk 5 above)
Comment (PascalinDe) [reply web/frontend/tests/json/evoting/forms/canceled.json:63]: https://www.din.de/de/wdc-beuth:din21:5463877

jedenfalls nicht in Deutschland :woman_shrugging:

**Pull request merged**